scripts/src/javatests/org/oppia/android/scripts/todo/TodoCollectorTest.kt

```diff
@@ -135,67 +206,82 @@ class TodoCollectorTest {
       """.trimIndent()
     val testContent2 =
       """
-      # TODO(#10500): Test description
+      <!--
+      TODO(# 105)
+      -->
       """.trimIndent()
     val testContent3 =
       """
-      // TODO(#17800): test todo.
-      // TODO(    210)
+      // TODO(#1788888): some description 5.
+      
+      
+      # TODO(#210000): some description 6.
       """.trimIndent()
     val tempFile1 = tempFolder.newFile("testfiles/TempFile1.txt")
-    val tempFile2 = tempFolder.newFile("testfiles/TempFile2.bazel")
-    val tempFile3 = tempFolder.newFile("testfiles/TempFile3.kt")
+    val tempFile2 = tempFolder.newFile("testfiles/TempFile2.kt")
+    val tempFile3 = tempFolder.newFile("testfiles/TempFile3.sh")
     tempFile1.writeText(testContent1)
     tempFile2.writeText(testContent2)
     tempFile3.writeText(testContent3)
 
-    val collectedTodos = TodoCollector.collectTodos(retrieveTestFilesDirectoryPath())
-    assertThat(collectedTodos).hasSize(5)
-    assertThat(collectedTodos).contains(
+    val allTodos = TodoCollector.collectTodos(retrieveTestFilesDirectoryPath())
+    val correctlyFormattedTodos = TodoCollector.collectCorrectlyFormattedTodos(allTodos)
+    assertThat(correctlyFormattedTodos).hasSize(5)
+    assertThat(correctlyFormattedTodos).contains(
       Todo(
         filePath = tempFile1.toString(),
         lineNumber = 1,
-        lineContent = "# TODO (#121): test todo."
+        lineContent = "// TODO(#12111): some description 1."
       )
     )
-    assertThat(collectedTodos).contains(
+    assertThat(correctlyFormattedTodos).contains(
       Todo(
         filePath = tempFile1.toString(),
         lineNumber = 2,
-        lineContent = "<!--TODO(#101)-->"
+        lineContent = "# TODO(#110000): some description 2."
       )
     )
-    assertThat(collectedTodos).contains(
+    assertThat(correctlyFormattedTodos).contains(
       Todo(
-        filePath = tempFile2.toString(),
-        lineNumber = 1,
-        lineContent = "# TODO(#10500): Test description"
+        filePath = tempFile1.toString(),
+        lineNumber = 3,
+        lineContent = "<!-- TODO(#1011010): some description 3. -->"
       )
     )
-    assertThat(collectedTodos).contains(
+    assertThat(correctlyFormattedTodos).contains(
       Todo(
         filePath = tempFile3.toString(),
         lineNumber = 1,
-        lineContent = "// TODO(#17800): test todo."
+        lineContent = "// TODO(#1788888): some description 5."
       )
     )
-    assertThat(collectedTodos).contains(
+    assertThat(correctlyFormattedTodos).contains(
       Todo(
         filePath = tempFile3.toString(),
-        lineNumber = 2,
-        lineContent = "// TODO(    210)"
+        lineNumber = 4,
+        lineContent = "# TODO(#210000): some description 6."
       )
     )
   }
 
+  @Test
+  fun testTodoCollector_parseIssueNumber_correctIssueNumberShouldBeObtained() {
+    val parsedIssueNumber = TodoCollector.parseIssueNumberFromTodo(
```

Comment:
Suggest also adding tests for parsing improper TODOs (in whatever scenarios you can think of).

Reply:
Done.